nav2_motion_primitives/src/back_up.cpp

```diff
@@ -66,46 +65,34 @@ nav2_tasks::TaskStatus BackUp::onCycleUpdate(nav2_tasks::BackUpResult & result)
   return status;
 }
 
-nav2_tasks::TaskStatus BackUp::timedBackup()
+
+nav2_tasks::TaskStatus BackUp::controlledBackup()
 {
-  // Output control command
-  geometry_msgs::msg::Twist cmd_vel;
+  auto current_odom_pose = std::make_shared<nav_msgs::msg::Odometry>();
 
-  // TODO(orduno): #423 assuming robot was moving fwd when it got stuck
-  //               fixed speed
-  cmd_vel.linear.x = -0.025;
+  if (!robot_->getOdometry(current_odom_pose)) {
```

Comment:
It seems unnecessary for getOdometry to take a SharedPtr. It should just take a simple pointer. That way the client code could use a simple local variable instead. That is, `current_odom_pose` could be a local variable, which seems the most desirable.

Reply:
Will address this on a different PR to also change getCurrentPose and getGlobalLocalizationPose.